pkg/cmdutils/config.go

```diff
@@ -158,42 +143,39 @@ func Exists(path string) bool {
 	return true
 }
 
-func (c *ClusterConfig) DecodeContext() *Config {
+func readConfigFile() (*Config, error) {
 	cfg := NewConfig()
 
 	defaultPath := fmt.Sprintf("%s/.config/pulsar/config", utils.HomeDir())
 	if !Exists(defaultPath) {
-		return nil
+		return nil, nil
 	}
 
 	content, err := ioutil.ReadFile(defaultPath)
 	if err != nil {
-		return nil
+		return nil, err
 	}
 
 	err = yaml.Unmarshal(content, &cfg)
 	if err != nil {
-		return nil
+		return nil, err
 	}
 
-	return cfg
+	return cfg, nil
 }
 
-func (c *ClusterConfig) Client(version common.APIVersion) pulsar.Client {
-	c.PulsarAPIVersion = version
-
-	ctxConf := c.DecodeContext()
+func (c *ClusterConfig) ApplyContext(ctxConf *Config, contextName *string) {
```

Comment:
This function is enhanced to allow a named context to be specified, or null to not load from any context.  This is important for the `pulsarctl context set` command.